aten/src/ATen/native/cpu/Loops.h

```diff
@@ -40,6 +40,13 @@ static inline bool is_unary_contiguous(const int64_t* strides) {
          strides[1] == sizeof(typename traits::arg1_t);
 }
 
+// output is contiguous, arg1 is scalar
+template <typename traits>
+static inline bool is_unary_contiguous_s1(const int64_t* strides) {
```

Comment:
Might this be more accurately described by a nullary operation as about to be added via https://github.com/pytorch/pytorch/pull/18876

Reply:
Yes, and it will save me from the writing vectorized loop for scalar case. Right now it is unnecessary as we never broadcast during the copy. But I look forward to both PRs to be landed to complete vec.

Reply:
I should add that the broadcasting happens  in between `Tensor::copy_` and `Tensor::s_copy_` by some auto-generated code. The copy code doesn't handle it explicitly. (Maybe that's what you meant?)

A lot of the functions that use `TensorIterator` no longer use this pattern (there's no longer an `s_add` or `s_mul`) because `TensorIterator` handles broadcasting and shape checks. Eventually we can do that for `copy_` (it'll reduce some boilerplate) but that will come later.

Reply:
> I should add that the broadcasting happens in between `Tensor::copy_` and `Tensor::s_copy_` by some auto-generated code. The copy code doesn't handle it explicitly. (Maybe that's what you meant?)

Yes. It is actually happening inside of the `_copy` by
```
std::tie(b_src) = expand_inplace(self, src, "copy");
```

> A lot of the functions that use `TensorIterator` no longer use this pattern (there's no longer an `s_add` or `s_mul`) because `TensorIterator` handles broadcasting and shape checks. Eventually we can do that for `copy_` (it'll reduce some boilerplate) but that will come later.

Agree, as soon as I check transpose edge case (as I suspect it to be slower than vectorized TensorIterator) we can remove resize and leave broadcasting to TensorIterator.